hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/utils/SystemContractUtils.java

```diff
@@ -18,16 +18,35 @@
 
 import static com.hedera.node.app.service.contract.impl.utils.ConversionUtils.tuweniToPbjBytes;
 
+import com.google.common.primitives.Longs;
 import com.hedera.hapi.node.base.ContractID;
 import com.hedera.hapi.node.contract.ContractFunctionResult;
+import com.hedera.node.app.service.contract.impl.state.ProxyEvmAccount;
+import com.hedera.node.app.service.contract.impl.state.ProxyWorldUpdater;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Arrays;
 import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.evm.frame.MessageFrame;
 
 /**
  * Utilities for system contracts.
  */
 public final class SystemContractUtils {
 
+    /*
+    TODO: here is how contractId is set in mono:
+    PrngSystemPrecompiledContract.computePrecompile > createSuccessfulChildRecord >
+    addContractCallResultToRecord > PrecompileUtils.addContractCallResultToRecord.
+    For the contractId in EvmFnResult is passed HTS_PRECOMPILE_MIRROR_ENTITY_ID - which is using the
+    HTC contract address(0x167). This seems like a bug to me.
+    Is this how it's suppose to work? Should I fix it in mono or should I just mimic it here and
+    create a story to later fix it?
+     */
+    public static final String HTS_PRECOMPILED_CONTRACT_ADDRESS = "0x167";
+    public static final ContractID HTS_PRECOMPILE_MIRROR_ID = contractIdFromEvmAddress(
+            Address.fromHexString(HTS_PRECOMPILED_CONTRACT_ADDRESS).toArrayUnsafe());
```

Comment:
Thoughts?

Reply:
Ah we can fix later imo, since we are still aiming to do "differential testing" between mono- and mod-service record streams.